lib/tasks/scrape_paws_data.rake

```diff
@@ -21,42 +21,43 @@ task({ :scrape_paws_data => :environment}) do
     pet.children_competability_ranking = if pet_parsed_page.css('.children .rating_default').children.to_s == "UNKNOWN" || pet_parsed_page.css('.children .rating_default').children.to_s == ""
-        then nil 
+         nil 
         else pet_parsed_page.css('.children .rating_default').children.to_s.split("")[14].to_i 
       end
     pet.activity_competability_ranking = if pet_parsed_page.css('.activity .rating_default').children.to_s == "UNKNOWN" || pet_parsed_page.css('.activity .rating_default').children.to_s == "" 
-        then nil 
+         nil 
         else pet_parsed_page.css('.activity .rating_default').children.to_s.split("")[14].to_i 
       end
     pet.cat_competability_ranking = if pet_parsed_page.css('.cats .rating_default').children.to_s == "UNKNOWN" || pet_parsed_page.css('.cats .rating_default').children.to_s == ""
-       then nil 
+        nil 
        else pet_parsed_page.css('.cats .rating_default').children.to_s.split("")[14].to_i 
       end
     pet.dog_competability_ranking = if pet_parsed_page.css('.dogs .rating_default').children.to_s == "UNKNOWN" || pet_parsed_page.css('.dogs .rating_default').children.to_s == "" 
-       then nil 
+        nil 
        else pet_parsed_page.css('.dogs .rating_default').children.to_s.split("")[14].to_i 
       end
     pet.enrichment_competability_ranking = if pet_parsed_page.css('.enrichment .rating_default').children.to_s == "UNKNOWN" || pet_parsed_page.css('.enrichment .rating_default').children.to_s == ""
-       then nil 
+        nil 
        else pet_parsed_page.css('.enrichment .rating_default').children.to_s.split("")[14].to_i 
       end
     pet.home_alone_competability_ranking = if pet_parsed_page.css('.home_alone .rating_default').children.to_s == "UNKNOWN" || pet_parsed_page.css('.home_alone .rating_default').children.to_s == ""
-       then nil 
+        nil 
        else pet_parsed_page.css('.home_alone .rating_default').children.to_s.split("")[14].to_i 
       end
     pet.human_socialability_competability_ranking = if pet_parsed_page.css('.human .rating_default').children.to_s == "UNKNOWN" || pet_parsed_page.css('.human .rating_default').children.to_s == "" 
-       then nil 
+        nil 
        else pet_parsed_page.css('.human .rating_default').children.to_s.split("")[14].to_i 
       end
-    pet.species = if link.include? "dog" then "Dog" else "Cat" end
+    pet.species = link.include?("dog") ? "Dog" : "Cat"
     pet.save 
     # create picture and tie to pet
-    picture = Picture.find_or_create_by(pet_id: pet.id)
-      img_url = pet_parsed_page.css('.lazyOwl').to_s.scan(/datasrc=.*\".*\"/)
-      img_url = img_url.to_s.scan(/https:.*jpg/).first
-      picture.image = img_url
-      picture.pet_id = pet.id 
-    picture.save
+    pet_parsed_page.css('.lazyOwl').each do |url|
+      unless url.values.first.nil?
+        img_url = url.values.first
+        picture = Picture.find_or_create_by(image: img_url)
+        pet.pictures.create(image: img_url)
```

Comment:
Both of these lines are creating a Picture record using the `img_url`. I think you only need one.

If you use the later approach you can use [CarrierWave's `remote_image_url` method](https://github.com/carrierwaveuploader/carrierwave/wiki/How-to:-Upload-remote-image-urls-to-your-seedfile)

```suggestion
        pet.pictures.create(remote_image_url: img_url)
```

Reply:
that worked! thanks for the feedback there